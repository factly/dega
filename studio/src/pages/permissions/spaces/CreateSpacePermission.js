import React from 'react';
import SpacePermissionCreateForm from './components/PermissionForm';
import { useDispatch } from 'react-redux';
import { addSpacePermission } from '../../../actions/spacePermissions';

import useNavigation from '../../../utils/useNavigation';

function CreateSpacePermission() {
  const history = useNavigation();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpacePermission(values)).then(() => history('/admin/permissions/spaces'));
  };
  return <SpacePermissionCreateForm onCreate={onCreate} />;
}

export default CreateSpacePermission;
