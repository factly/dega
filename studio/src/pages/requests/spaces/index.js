import React from 'react';
import SpaceRequestList from './components/RequestList';
import { Space, Button, Row } from 'antd';
import { Link } from 'react-router-dom';
import { PlusOutlined } from '@ant-design/icons';
import { Helmet } from 'react-helmet';

function SpaceRequests() {
  return (
    <Space direction="vertical">
      <Helmet title={'Space Requests'} />
      <Row gutter={16} justify="end">
        <Link key="1" to="/admin/requests/spaces/create">
          <Button
            icon={<PlusOutlined />}
            type="primary">Create</Button>
        </Link>
      </Row>

      <SpaceRequestList />
    </Space>
  );
}

export default SpaceRequests;
