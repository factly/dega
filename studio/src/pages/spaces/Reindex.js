import React from 'react';
import { Row, Col, Button, Form, Select, Typography, Space, ConfigProvider } from 'antd';
import { useSelector, useDispatch } from 'react-redux';
import { reindex, reindexSpace } from '../../actions/meiliReindex';

const Reindex = () => {
  const { Title } = Typography;

  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const [spaceOption, setSpaceOption] = React.useState();
  const [status, setStatus] = React.useState();
  const [spaceSelected, setSpaceSelected] = React.useState(false);

  const superOrg = useSelector(({ admin }) => {
    return admin.organisation;
  });

  const { orgId, orgs, spaces } = useSelector((state) => {
    const { selected, orgs, loading } = state.spaces;
    if (selected > 0) {
      const space = state.spaces.details[selected];
      const orgId = space.organisation_id;
      const org = orgs.find((org) => org.id === orgId);
      const role = org.permission.role;

      return {
        loading: loading,
        role: role,
        orgId: orgId,
        spaces: state.spaces.details,
        orgs: superOrg.is_admin ? state.spaces.orgs : [org],
      };
    }
    return { orgId: 0, orgs: [], spaces: [] };
  });
  const [organisationId, setOrganisationId] = React.useState(orgId);

  React.useEffect(() => {
    if (orgs.length > 0) {
      const org = orgs.find((org) => org.id === organisationId);
      const newSpaces = org.spaces.map((space) => spaces[space]);
      setSpaceOption(newSpaces);
    }
  }, [organisationId]);

  const handleorgChange = (orgId) => {
    form.setFieldsValue({
      space_id: null,
    });
    setOrganisationId(orgId);
  };
  const handleReindex = () => {
    dispatch(reindex());
  };
  const handleSpaceReindex = (id) => {
    if (id) {
      dispatch(reindexSpace(id));
    } else {
      setStatus('error');
    }
  };
  return (
    <ConfigProvider
      theme={{
        components: {
          Form: {
            marginLG: 8,
          },
        },
      }}
    >
      <Space direction="vertical">
        <Form form={form} name="reindex" layout="vertical">
          <Row justify="start" style={{ maxWidth: '1200px', width: '100%', margin: '0 auto' }}>
            <Col span={24}>
              <Row justify="end">
                <Button
                  type="primary"
                  onClick={() => handleSpaceReindex(form.getFieldValue('space_id'))}
                  disabled={!spaceSelected}
                >
                  Reindex
                </Button>
              </Row>
              <Row
                gutter={40}
                justify="start"
                style={{ marginBottom: '1rem' }}
              >
                <Col md={8} xs={24}>
                  {orgs && orgs.length > 0 ? (
                    <Form.Item name="organisation_id" label="Organisation">
                      <Select
                        allowClear
                        bordered
                        listHeight={128}
                        defaultValue={[orgId]}
                        onChange={(value) => handleorgChange(value)}
                      >
                        {orgs.map((item) => (
                          <Select.Option value={item.id} key={item.id}>
                            {item.title}
                          </Select.Option>
                        ))}
                      </Select>
                    </Form.Item>
                  ) : null}

                  {spaceOption && spaceOption.length > 0 ? (
                    <Form.Item name="space_id" label="Space" validateStatus={status}>
                      <Select
                        allowClear
                        bordered
                        listHeight={128}
                        defaultValue={[]}
                        placeholder="Select Space"
                        onChange={() => setStatus()}
                        onSelect={() => setSpaceSelected(true)}
                        onClear={() => setSpaceSelected(false)}
                      >
                        {spaceOption.map((item) => (
                          <Select.Option value={item.id} key={item.id}>
                            {item.name}
                          </Select.Option>
                        ))}
                      </Select>
                    </Form.Item>
                  ) : null}
                </Col>
              </Row>
            </Col>
          </Row>
        </Form>
      </Space>
    </ConfigProvider>
  );
};

export default Reindex;
