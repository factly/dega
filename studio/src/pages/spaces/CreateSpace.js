import React from 'react';
import SpaceCreateForm from './components/SpaceCreateForm';
import { useDispatch } from 'react-redux';
import { addSpace, getSpaces } from '../../actions/spaces';

import { Helmet } from 'react-helmet';
import useNavigation from '../../utils/useNavigation';

function CreateSpace() {
  const history = useNavigation();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpace(values)).then(() => {
      dispatch(getSpaces());
      history('/admin/spaces');
    });
  };
  return (
    <>
      <Helmet title={'Create Space'} />
      <SpaceCreateForm onCreate={onCreate} />
    </>
  );
}

export default CreateSpace;
