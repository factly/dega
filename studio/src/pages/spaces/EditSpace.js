import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Skeleton } from 'antd';
import { updateSpace } from '../../actions/spaces';
import SpaceEditForm from './components/SpaceEditForm';
import RecordNotFound from '../../components/ErrorsAndImage/RecordNotFound';
import { Helmet } from 'react-helmet';
import useNavigation from '../../utils/useNavigation';

function EditSpace() {
  const history = useNavigation();
  const { id } = useParams();
  const dispatch = useDispatch();

  const { space, loading } = useSelector((state) => {
    return {
      space: state.spaces.details[id],
      loading: state.spaces.loading,
    };
  });

  const onCreate = (values) => {
    dispatch(updateSpace({ ...space, ...values })).then(() => history(`/admin/spaces`));
  };

  if (loading) return <Skeleton />;

  if (!space) {
    return <RecordNotFound />;
  }

  return (
    <>
      <Helmet title={`${space?.name} - Edit Space`} />
      <SpaceEditForm onCreate={onCreate} data={space} />
    </>
  );
}

export default EditSpace;
