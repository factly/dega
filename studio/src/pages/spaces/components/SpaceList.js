import React from 'react';
import { ConfigProvider, Modal, Button, Table, Typography } from 'antd';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { deleteSpace, getSpaces } from './../../../actions/spaces';
import { spaceSelector } from '../../../selectors/spaces';
import { DeleteOutlined } from '@ant-design/icons';
import useNavigation from '../../../utils/useNavigation';

function SpaceList() {
  const dispatch = useDispatch();
  const { spaces, loading } = useSelector(spaceSelector);

  const [modalOpen, setModalOpen] = React.useState(false);
  const [deleteItemId, setDeleteItemId] = React.useState(null);

  const history = useNavigation();

  const fetchSpaces = () => {
    dispatch(getSpaces());
  };

  const columns = [
    {
      title: 'ID',
      dataIndex: 'id',
      key: 'id',
      width: 200,
      render: (_, record) => {
        return (
          <Link
            style={{
              marginRight: 8,
            }}
            to={`/admin/spaces/${record.id}/edit`}
          >
            <Typography.Text style={{ fontSize: '1rem', color: '#101828' }}>
              {record.id}
            </Typography.Text>
          </Link>
        );
      },
    },
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
      width: 200,
      onCell: () => {
        return {
          style: {
            minWidth: '200px',
          },
        };
      },
      render: (_, record) => {
        return (
          <Link
            style={{
              marginRight: 8,
            }}
            to={`/admin/spaces/${record.id}/edit`}
          >
            <Typography.Text style={{ fontSize: '1rem', color: '#101828' }} strong>
              {record.name}
            </Typography.Text>
          </Link>
        );
      },
    },
    {
      title: 'Site Address',
      dataIndex: 'site_address',
      key: 'site_address',
      width: 200,
      onCell: () => {
        return {
          style: {
            minWidth: '200px',
          },
        };
      },
      render: (_, record) => {
        return (
          <Typography.Text style={{ fontSize: '1rem', color: '#101828' }} strong>
            {record.site_address}
          </Typography.Text>
        );
      },
    },
    {
      title: 'Site Title',
      dataIndex: 'site_title',
      key: 'site_title',
      width: 200,
      onCell: () => {
        return {
          style: {
            minWidth: '200px',
          },
        };
      },
      render: (_, record) => {
        return (
          <Typography.Text style={{ fontSize: '1rem', color: '#101828' }} strong>
            {record.site_title}
          </Typography.Text>
        );
      },
    },
    {
      title: 'Tag line',
      dataIndex: 'tag_line',
      key: 'tag_line',
      width: 200,
      onCell: () => {
        return {
          style: {
            minWidth: '200px',
          },
        };
      },
      render: (_, record) => {
        return (
          <Typography.Text style={{ fontSize: '1rem', color: '#101828' }} strong>
            {record.tag_line}
          </Typography.Text>
        );
      },
    },
    {
      title: 'Action',
      dataIndex: 'operation',
      align: 'center',
      width: 150,
      onCell: () => {
        return {
          style: {
            minWidth: '150px',
          },
        };
      },
      render: (_, record) => {
        return (
          <ConfigProvider
            theme={{
              components: {
                Button: {
                  controlHeight: 35,
                  colorBorder: '#F2F2F2',
                  colorPrimaryHover: '#00000026',
                },
              },
            }}
          >
            <div style={{ display: 'flex', gap: '0.5rem' }}>
              <Button
                size="large"
                style={{
                  backgroundColor: 'red',
                  borderColor: 'red',
                  color: '#fff',
                }}
                onClick={(e) => {
                  e.stopPropagation();
                  setModalOpen(true);
                  setDeleteItemId(record.id);
                }}
                icon={<DeleteOutlined style={{ color: '#fff' }} />}
              />
            </div>
          </ConfigProvider>
        );
      },
    },
  ];

  return (
    <ConfigProvider
      theme={{
        components: {
          Typography: {
            colorText: '#101828',
          },
        },
      }}
    >
      <Table
        onRow={(record, rowIndex) => {
          return {
            onClick: (event) => {
              history(`/admin/spaces/${record.id}/edit`);
            },
            onMouseEnter: (event) => {
              document.body.style.cursor = 'pointer';
            },
            onMouseLeave: (event) => {
              document.body.style.cursor = 'default';
            },
          };
        }}
        // style={{ maxWidth: '100vw', overflowX: 'auto' }}
        scroll={{
          x: '1000',
        }}
        rowKey={'id'}
        dataSource={spaces}
        columns={columns}
        loading={loading}
      />
      <Modal
        open={modalOpen}
        closable={false}
        centered
        width={311}
        className="delete-modal-container"
        style={{
          borderRadius: '18px',
        }}
        onOk={(e) => {
          e.stopPropagation();
          dispatch(deleteSpace(deleteItemId)).then(() => fetchSpaces());
          setModalOpen(false);
          setDeleteItemId(null);
        }}
        cancelButtonProps={{ type: 'text', style: { color: '#000' } }}
        onCancel={(e) => {
          e.stopPropagation();
          setModalOpen(false);
          setDeleteItemId(null);
        }}
      >
        <Typography.Text style={{ fontSize: '1rem', color: '#101828' }} strong>
          Are you sure you want to delete this space?
        </Typography.Text>
      </Modal>
    </ConfigProvider>
  );
}

export default SpaceList;
