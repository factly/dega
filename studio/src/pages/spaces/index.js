import React from 'react';
import { Space, Button, Row, Col, ConfigProvider } from 'antd';
import { Link } from 'react-router-dom';
import SpaceList from './components/SpaceList';
import { useSelector } from 'react-redux';
import { Helmet } from 'react-helmet';

function Spaces() {
  const { role } = useSelector((state) => {
    const { selected } = state.spaces;

    if (selected !== '') {
      const space = state.spaces.details[selected];
      const role = space.org_role;
      return {
        role: role,
      };
    }
    return { role: 'member' };
  });

  return (
    <ConfigProvider
      theme={{
        components: {
          Button: {
            controlHeight: 35,
            colorBorder: '#1890FF',
            colorPrimaryHover: '#00000026',
            colorText: '#1890FF',
            paddingXS: '2px, 16px, 2px, 16px',
          },
        },
      }}
    >
      <Space direction="vertical">
        <Helmet title={'Spaces'} />
        <Row gutter={16} justify="end">
          {role === 'admin' ? (
            <Col>
              <Link key="2" to="/settings/advanced/reindex">
                <Button>Reindex</Button>
              </Link>
            </Col>
          ) : null}
          <Link key="1" to="/spaces/create">
            <Button type="primary">New Space</Button>
          </Link>
        </Row>
        <SpaceList />
      </Space>
    </ConfigProvider>
  );
}

export default Spaces;
