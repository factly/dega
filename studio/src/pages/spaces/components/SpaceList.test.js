import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Popconfirm, Button } from 'antd';

import '../../../matchMedia.mock';
import SpaceList from './SpaceList';
import { getSpaces, deleteSpace } from '../../../actions/spaces';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock('../../../actions/spaces', () => ({
  getSpaces: jest.fn(),
  deleteSpace: jest.fn(),
}));

describe('Spaces List component', () => {
  let store;
  let mockedDispatch;
  const space = {
    id: 1,
    name: 'space 1',
    site_address: 'site_address',
    site_title: 'site_title',
    tag_line: 'tag_line',
  };

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      useSelector.mockImplementation((state) => ({}));
      const tree = renderer.create(<SpaceList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component when loading', () => {
      useSelector.mockImplementation((state) => ({
        spaces: [],
        loading: true,
      }));
      const tree = renderer.create(<SpaceList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component with spaces', () => {
      useSelector.mockImplementation((state) => ({
        spaces: [space],
        loading: false,
      }));

      let component;
      act(() => {
        component = renderer.create(
          <Router>
            <SpaceList />
          </Router>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      expect(useSelector).toHaveBeenCalled();
      expect(useSelector).toHaveReturnedWith({
        spaces: [space],
        loading: false,
      });
    });
  });
  describe('component testing', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should delete space', () => {
      useSelector.mockImplementation((state) => ({
        spaces: [space],
        total: 1,
        loading: false,
      }));

      const wrapper = mount(
        <Router>
          <SpaceList />
        </Router>,
      );
      const button = wrapper.find(Button).at(0);
      expect(button.text()).toEqual('');

      button.simulate('click');
      const popconfirm = wrapper.find(Popconfirm);
      popconfirm
        .findWhere((item) => item.type() === 'button' && item.text() === 'OK')
        .simulate('click');
      setTimeout(() => {
        expect(deleteSpace).toHaveBeenCalled();
        expect(deleteSpace).toHaveBeenCalledWith(1);
        expect(getSpaces).toHaveBeenCalledWith({ page: 1 });
      });
    });
    it('should edit space', () => {
      useSelector.mockImplementation((state) => ({
        spaces: [space],
        total: 1,
        loading: false,
      }));

      const wrapper = mount(
        <Router>
          <SpaceList />
        </Router>,
      );
      const link = wrapper.find(Link).at(0);
      expect(link.text()).toEqual('space 1');
      expect(link.prop('to')).toEqual('/admin/spaces/1/edit');
    });
    it('should have no delete and edit buttons', () => {
      useSelector.mockImplementation((state) => ({}));

      const wrapper = mount(
        <Router>
          <SpaceList />
        </Router>,
      );

      const button = wrapper.find(Button);
      expect(button.length).toEqual(0);
    });
  });
});
