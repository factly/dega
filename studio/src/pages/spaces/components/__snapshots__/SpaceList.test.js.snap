// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spaces List component snapshot testing should match component when loading 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div>
      <div
        className="ant-spin ant-spin-spinning"
      >
        <span
          className="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
    <div
      className="ant-spin-container ant-spin-blur"
    >
      <div
        className="ant-table ant-table-bordered ant-table-empty ant-table-has-fix-right"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "width": "8%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "12%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Id
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Site Address
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Site Title
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Tag line
                  </th>
                  <th
                    className="ant-table-cell ant-table-cell-fix-right ant-table-cell-fix-right-first"
                    colSpan={null}
                    rowSpan={null}
                    style={
                      Object {
                        "position": "sticky",
                        "right": 0,
                        "textAlign": "center",
                      }
                    }
                  >
                    Action
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-placeholder"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                >
                  <td
                    className="ant-table-cell"
                    colSpan={6}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <div
                      className="ant-empty ant-empty-normal"
                    >
                      <div
                        className="ant-empty-image"
                      >
                        <svg
                          className="ant-empty-img-simple"
                          height="41"
                          viewBox="0 0 64 41"
                          width="64"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                            transform="translate(0 1)"
                          >
                            <ellipse
                              className="ant-empty-img-simple-ellipse"
                              cx="32"
                              cy="33"
                              rx="32"
                              ry="7"
                            />
                            <g
                              className="ant-empty-img-simple-g"
                              fillRule="nonzero"
                            >
                              <path
                                d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                              />
                              <path
                                className="ant-empty-img-simple-path"
                                d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                              />
                            </g>
                          </g>
                        </svg>
                      </div>
                      <div
                        className="ant-empty-description"
                      >
                        No Data
                      </div>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Spaces List component snapshot testing should match component with spaces 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered ant-table-has-fix-right"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "width": "8%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "12%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Id
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Site Address
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Site Title
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Tag line
                  </th>
                  <th
                    className="ant-table-cell ant-table-cell-fix-right ant-table-cell-fix-right-first"
                    colSpan={null}
                    rowSpan={null}
                    style={
                      Object {
                        "position": "sticky",
                        "right": 0,
                        "textAlign": "center",
                      }
                    }
                  >
                    Action
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    1
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <a
                      href="/admin/spaces/1/edit"
                      onClick={[Function]}
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    >
                      space 1
                    </a>
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    site_address
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    site_title
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    tag_line
                  </td>
                  <td
                    className="ant-table-cell ant-table-cell-fix-right ant-table-cell-fix-right-first"
                    colSpan={null}
                    rowSpan={null}
                    style={
                      Object {
                        "position": "sticky",
                        "right": 0,
                        "textAlign": "center",
                      }
                    }
                  >
                    <button
                      className="ant-btn ant-btn-danger ant-btn-icon-only"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="delete"
                        className="anticon anticon-delete"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="delete"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                          />
                        </svg>
                      </span>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Spaces List component snapshot testing should render the component 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered ant-table-empty ant-table-has-fix-right"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "width": "8%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "12%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "width": 150,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Id
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Site Address
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Site Title
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Tag line
                  </th>
                  <th
                    className="ant-table-cell ant-table-cell-fix-right ant-table-cell-fix-right-first"
                    colSpan={null}
                    rowSpan={null}
                    style={
                      Object {
                        "position": "sticky",
                        "right": 0,
                        "textAlign": "center",
                      }
                    }
                  >
                    Action
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-placeholder"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                >
                  <td
                    className="ant-table-cell"
                    colSpan={6}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <div
                      className="ant-empty ant-empty-normal"
                    >
                      <div
                        className="ant-empty-image"
                      >
                        <svg
                          className="ant-empty-img-simple"
                          height="41"
                          viewBox="0 0 64 41"
                          width="64"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                            transform="translate(0 1)"
                          >
                            <ellipse
                              className="ant-empty-img-simple-ellipse"
                              cx="32"
                              cy="33"
                              rx="32"
                              ry="7"
                            />
                            <g
                              className="ant-empty-img-simple-g"
                              fillRule="nonzero"
                            >
                              <path
                                d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                              />
                              <path
                                className="ant-empty-img-simple-path"
                                d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                              />
                            </g>
                          </g>
                        </svg>
                      </div>
                      <div
                        className="ant-empty-description"
                      >
                        No Data
                      </div>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
